File: frontend/src/api/llmops.ts
import apiClient from './index'

export interface LLMOpsMetrics {
  all_time: {
    total_calls: number
    total_tokens: number
    total_cost_usd: number
    avg_latency_ms: number
    success_rate: number
  }
  today: {
    calls: number
    tokens: number
    cost_usd: number
    avg_latency_ms: number
    success_rate: number
  }
}

export interface DailyStats {
  date: string
  calls: number
  tokens: number
  cost_usd: number
  avg_latency_ms: number
  success_rate: number
}

export interface RecentCall {
  call_id: string
  model: string
  tokens: number
  latency_ms: number
  cost_usd: number
  success: boolean
  error?: string
  timestamp: string
}

export const llmopsApi = {
  getMetrics: (days = 7) =>
    apiClient.get<LLMOpsMetrics>('/llmops/metrics', { params: { days } }),

  getDailyStats: (days = 30) =>
    apiClient.get<{ daily_stats: DailyStats[] }>('/llmops/daily', { params: { days } }),

  getRecentCalls: (limit = 50) =>
    apiClient.get<{ recent_calls: RecentCall[] }>('/llmops/recent', { params: { limit } }),
}

File: backend/app/api/llmops.py
"""
backend/app/api/llmops.py
LLMOps 监控 API - 模型调用统计、成本追踪与趋势分析
"""
import logging

from fastapi import APIRouter, Query

from app.llmops.monitor import get_llmops_monitor

logger = logging.getLogger(__name__)
router = APIRouter()


@router.get("/llmops/metrics")
async def get_metrics(days: int = Query(7, ge=1, le=90, description="统计天数")):
    """获取 LLMOps 核心监控指标（汇总 + 今日）"""
    monitor = get_llmops_monitor()
    return monitor.get_metrics(days=days)


@router.get("/llmops/daily")
async def get_daily_stats(
    days: int = Query(30, ge=1, le=90, description="统计天数"),
):
    """获取按日统计的调用趋势数据"""
    monitor = get_llmops_monitor()
    return {"daily_stats": monitor.get_daily_stats(days=days)}


@router.get("/llmops/recent")
async def get_recent_calls(
    limit: int = Query(50, ge=1, le=200, description="返回条数"),
):
    """获取最近的LLM调用记录"""
    monitor = get_llmops_monitor()
    return {"recent_calls": monitor.get_recent_calls(limit=limit)}
